src/modules/fancyzones/lib/FancyZones.cpp

```diff
@@ -323,36 +324,59 @@ FancyZones::VirtualDesktopInitialize() noexcept
     PostMessage(m_window, WM_PRIV_VD_INIT, 0, 0);
 }
 
+bool FancyZones::MoveWindowIntoLastKnownZone(HWND window, winrt::com_ptr<IZoneWindow> zoneWindow) {
+    const auto activeZoneSet = zoneWindow->ActiveZoneSet();
+    if (activeZoneSet)
+    {
+        wil::unique_cotaskmem_string guidString;
+        if (SUCCEEDED(StringFromCLSID(activeZoneSet->Id(), &guidString)))
+        {
+            auto& fancyZonesData = JSONHelpers::FancyZonesDataInstance();
+            std::vector<int> zoneIndexSet = fancyZonesData.GetAppLastZoneIndexSet(window, zoneWindow->UniqueId(), guidString.get());
+            if (zoneIndexSet.size() &&
+                !fancyZonesData.IsAnotherWindowOfApplicationInstanceZoned(window, zoneWindow->UniqueId()))

```

Comment:
This is wrong, because it could cause a child window to be zoned to a different monitor from the parent window.

Reply:
The way I see it we need to find within same desktop, is another window of application instance zoned. If it's zoned on another monitor than the one we are currently considering, we should abort.

Reply:
The design needs to be changed because this approach is not working.
To fix it do the following:
 - rename `MoveWindowIntoLastKnownZone` to `GetZoneSetFromCursor`
 - move this code into:
```
POINT cursorPosition{};
        GUID desktopId{};
        if (GetCursorPos(&cursorPosition) &&
            VirtualDesktopUtils::GetWindowDesktopId(window, &desktopId))
        {
            HMONITOR monitor = MonitorFromPoint(cursorPosition, MONITOR_DEFAULTTOPRIMARY);
            auto zoneWindow = m_workAreaHandler.GetWorkArea(desktopId, monitor);
            if (zoneWindow && MoveWindowIntoLastKnownZone(window, zoneWindow))
            {
                return;
            }
        }
```
 - `GetZoneSetFromCursor()` will not call `IsAnotherWindowOfApplicationInstanceZoned()`
 - in `ProcessNewWindow()` if `GetZoneSetFromCursor()` returns null, call:
```
auto zoneWindow = m_workAreaHandler.GetWorkArea(window);
```
 - then call `IsAnotherWindowOfApplicationInstanceZoned()`

Reply:
`IsAnotherWindowOfApplicationInstanceZoned(HWND window, const std::wstring& desktopId)`
and check on same desktop, accross monitors

Reply:
In a nutshell, first check if there is a zoneset for the current monitor, if not, use the default zoneset.
At this point if you have a valid zoneset, do the final check calling `IsAnotherWindowOfApplicationInstanceZoned` and then call `MoveWindowIntoZoneByIndexSet`

Reply:
I've updated PR with new iteration.